tests/testfunctions/snapshots/logs/helloApiGateway_Java11.log

```diff
@@ -0,0 +1,46 @@
+2021-03-19 19:22:16 <2a3114f0-8856-4e10-a452-f85a760698fb> INFO  com.serverless.Handler:20 - received: {resource: /{proxy+},path: /test/hello,httpMethod: GET,headers: {Accept=text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8, Accept-Encoding=gzip, deflate, lzma, sdch, br, Accept-Language=en-US,en;q=0.8, CloudFront-Forwarded-Proto=https, CloudFront-Is-Desktop-Viewer=true, CloudFront-Is-Mobile-Viewer=false, CloudFront-Is-SmartTV-Viewer=false, CloudFront-Is-Tablet-Viewer=false, CloudFront-Viewer-Country=US, Host=wt6mne2s9k.execute-api.us-west-2.amazonaws.com, Upgrade-Insecure-Requests=1, User-Agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.82 Safari/537.36 OPR/39.0.2256.48, Via=1.1 fb7cca60f0ecd82ce07790c9c5eef16c.cloudfront.net (CloudFront), X-Amz-Cf-Id=nBsWBOrSHMgnaROZJK1wGCZ9PcRcSpq_oSXZNQwQ10OTZL4cimZo3g==, X-Forwarded-For=192.168.100.1, 192.168.1.1, X-Forwarded-Port=443, X-Forwarded-Proto=https},queryStringParameters: {name=me},pathParameters: {proxy=hello},stageVariables: {stageVarName=stageVarValue},requestContext: {accountId: 123456789012,resourceId: us4z18,stage: test,requestId: 41b45ea3-70b5-11e6-b7bd-69b5aaebc7d9,identity: {cognitoIdentityPoolId: ,accountId: ,cognitoIdentityId: ,caller: ,apiKey: ,sourceIp: 192.168.100.1,eventTriggerConfigId: ,cognitoAuthenticationProvider: ,userArn: ,userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.82 Safari/537.36 OPR/39.0.2256.48,user: ,},resourcePath: /{proxy+},httpMethod: GET,apiId: wt6mne2s9k,},}
+END RequestId: XXXX
+REPORT RequestId: XXXX	Duration: XXXX ms	Billed Duration: XXXX ms	Memory Size: 1024 MB	Max Memory Used: XXXX MB	Init Duration: XXXX ms	
+{"traces":[[{"service":"aws.lambda","name":"aws.lambda","resource":"Java-integration-test-dev-helloApiGateway_Java11","trace_id":"4f613f1405d29952","span_id":"158d1180ed2f4d12","parent_id":"0","start":1616181736704043436,"duration":57254254,"error":0,"metrics":{"_dd.agent_psr":1.0},"meta":{"function_arn":"arn:aws:lambda:sa-east-1:601427279990:function:java-integration-test-dev-helloapigateway_java11","runtime-id":"246bfcf8-4af7-4b91-98e1-aa7887c1f8e3","language":"jvm","thread.name":"main","request_id":"2a3114f0-8856-4e10-a452-f85a760698fb","datadog_lambda":"0.2.2","thread.id":"1","cold_start":"true","resource_names":"Java-integration-test-dev-helloApiGateway_Java11","function_version":"$LATEST"}}]]}START RequestId: XXXX Version: $LATEST
+{"m":"aws.lambda.enhanced.invocations","v":1.0,"t":["memorysize:1024","account_id:XXXX","resource:Java-integration-test-dev-helloApiGateway_Java11","functionname:Java-integration-test-dev-helloApiGateway_Java11","runtime:java11.0.9.1","region:sa-east-1","datadog_lambda:0.2.2","cold_start:false"],"e":1616181739}
+java.lang.NullPointerException: java.lang.NullPointerException
```

Comment:
Is this an intentional error? It looks like we might have caught a bug

Reply:
It took me a minute to figure out what is happening here. The test script is throwing all inputs at all handlers, even when there's a type mismatch between the input and the type the handler accepts (e.g. passing an SQS event to a handler that accepts an API Gateway proxy request event). I think that's what's happening here.

Not sure what the Lambda runtime's behavior is when the provided input doesn't match the handler's signature. My off the cuff guess is that the input is probably null, so we can add an extra layer of defense and check for null input. I'll verify this.